lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -3,29 +3,40 @@
  * Licensed under the MIT License.
  */
 
-import { AccountInfo, AuthenticationResult, Constants, RequestThumbprint, AuthError, PerformanceEvents, ServerError, InteractionRequiredAuthError, InProgressPerformanceEvent } from "@azure/msal-common";
-import { Configuration } from "../config/Configuration";
-import { DEFAULT_REQUEST, InteractionType, ApiId, CacheLookupPolicy, BrowserConstants } from "../utils/BrowserConstants";
-import { IPublicClientApplication } from "./IPublicClientApplication";
-import { RedirectRequest } from "../request/RedirectRequest";
+import { ITokenCache } from "../cache/ITokenCache";
+import { INavigationClient } from "../navigation/INavigationClient";
+import { AuthorizationCodeRequest } from "../request/AuthorizationCodeRequest";
 import { PopupRequest } from "../request/PopupRequest";
-import { ClientApplication } from "./ClientApplication";
+import { RedirectRequest } from "../request/RedirectRequest";
 import { SilentRequest } from "../request/SilentRequest";
-import { EventType } from "../event/EventType";
-import { BrowserAuthError } from "../error/BrowserAuthError";
-import { NativeAuthError } from "../error/NativeAuthError";
-import { NativeMessageHandler } from "../broker/nativeBroker/NativeMessageHandler";
-import { BrowserUtils } from "../utils/BrowserUtils";
+import { WrapperSKU } from "../utils/BrowserConstants";
+import { IPublicClientApplication } from "./IPublicClientApplication";
+import { IController } from "../controllers/IController";
+import { AuthenticationResult, PerformanceCallbackFunction, AccountInfo, Logger } from "@azure/msal-common";
+import { EndSessionRequest } from "../request/EndSessionRequest";
+import { SsoSilentRequest } from "../request/SsoSilentRequest";
+import { ControllerFactory } from "../controllers/ControllerFactory";
+import { StandardController } from "../controllers/StandardController";
+import { BrowserConfiguration, buildConfiguration, Configuration } from "../config/Configuration";
+import { version, name } from "../packageMetadata";
+import { StandardOperatingContext } from "../operatingcontext/StandardOperatingContext";
 
 /**
  * The PublicClientApplication class is the object exposed by the library to perform authentication and authorization functions in Single Page Applications
  * to obtain JWT tokens as described in the OAuth 2.0 Authorization Code Flow with PKCE specification.
  */
-export class PublicClientApplication extends ClientApplication implements IPublicClientApplication {
+export class PublicClientApplication implements IPublicClientApplication {
+
+    private controller: IController;
 
-    // Active requests
-    private activeSilentTokenRequests: Map<string, Promise<AuthenticationResult>>;
-    private astsAsyncMeasurement?: InProgressPerformanceEvent = undefined;
+    public static async createPublicClientApplication(configuration: Configuration): Promise<IPublicClientApplication> {
+
+        const factory = new ControllerFactory(configuration);
```

Comment:
Can we add some context here on what `StandardController` vs `MetaOSController` would mean? Also will these be added to our type docs?

Reply:
This is internal.  I told Hector I'd add some additional comments in the PRs actually adding support for Teams Apps.